src/goto-symex/renaming_level.cpp

```diff
@@ -77,6 +77,45 @@ void symex_level1t::restore_from(
   }
 }
 
+/// Allocates a fresh L2 name for the given L1 identifier, and makes it the
+/// latest generation on this path.
+void symex_level2t::increase_generation(
+  const irep_idt l1_identifier,
+  const ssa_exprt &lhs)
+{
+  INVARIANT(
+    global_names != nullptr, "Global level 2 naming map can't be null.");
+
+  current_names.emplace(l1_identifier, std::make_pair(lhs, 0));
+  global_names->emplace(l1_identifier, std::make_pair(lhs, 0));
+
+  increase_generation_if_exists(l1_identifier);
```

Comment:
I think there needs to be third, private, function that both `increase_generation` and `increase_generation_if_exists` invoke. That function should take iterators into `current_names` and `global_names`. As is, we first do an `emplace` here and then again do lookups inside `increase_generation_if_exists` (as well as a whole lot of redundant invariant checking).

Reply:
In the end I just duplicated the common code in each method as it was just two lines.